lib/sentry/integrations/oban/error_reporter.ex

```diff
@@ -33,18 +33,22 @@ defmodule Sentry.Integrations.Oban.ErrorReporter do
       end
 
     _ =
-      Sentry.capture_exception(exception,
-        stacktrace: stacktrace,
-        tags: %{oban_worker: job.worker, oban_queue: job.queue, oban_state: job.state},
-        fingerprint: [
-          inspect(exception.__struct__),
-          inspect(job.worker),
-          Exception.message(exception)
-        ],
-        extra:
-          Map.take(job, [:args, :attempt, :id, :max_attempts, :meta, :queue, :tags, :worker]),
-        integration_meta: %{oban: %{job: job}}
-      )
+      if is_exception(exception) do
+        Sentry.capture_exception(exception,
+          stacktrace: stacktrace,
+          tags: %{oban_worker: job.worker, oban_queue: job.queue, oban_state: job.state},
+          fingerprint: [
+            inspect(exception.__struct__),
+            inspect(job.worker),
+            Exception.message(exception)
+          ],
+          extra:
+            Map.take(job, [:args, :attempt, :id, :max_attempts, :meta, :queue, :tags, :worker]),
+          integration_meta: %{oban: %{job: job}}
+        )
+      else
+        Sentry.capture_message("Error with %s", interpolation_parameters: [exception])
+      end
```

Comment:
It would be more clear to have `reason` instead of `exception` here since it may not be an exception after all :)

Reply:
True! Updated, thanks!

Reply:
This is a wildly generic error message we got going on here 😄 I think it'd be really hard to debug as it's just a generic message plus there are no metadata about the job (which you have available).

This needs to be completely restructured to basically build shared options, and then use `Sentry.capture_message` or `Sentry.capture_exception` based on whether `reason` is an exception or not. Also, doesn't `:interpolation_parameters` require string parameters? If so, do `inspect(reason)` here.

Reply:
Also let's have the message be something like

```elixir
"Oban job #{job.worker} errored out: %s"
```

Btw the fingerprint option needs to be customized, if it's just reason use `inspect(reason)` insted of `Exception.message/1`